src/mixins/object_interactivity.mixin.ts

```diff
@@ -19,7 +20,9 @@ type TOCoord = IPoint & {
 
 type TControlSet = Record<string, Control>;
 
-export class InteractiveFabricObject extends FabricObject {
+export class InteractiveFabricObject<
+  T extends TObjectEvents = TObjectEvents
+> extends FabricObject<T> {
```

Comment:
made it generic for future use, when #8420 is merged I will move the text events to there so TS won't show they are possible to listen to on other objects (same for group)